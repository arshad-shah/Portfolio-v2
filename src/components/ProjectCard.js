import React from "react";
import { styled } from "@mui/material/styles";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardHeader from "@mui/material/CardHeader";
import { Chip, IconButton, Tooltip, Typography } from "@mui/material";
import { FiGithub, FiExternalLink } from "react-icons/fi";
import {
  SiCss3,
  SiHtml5,
  SiJava,
  SiJavascript,
  SiJquery,
  SiKotlin,
  SiMaterialui,
  SiNextdotjs,
  SiReact,
  SiRedux,
  SiSqlite,
  SiStyledcomponents,
  SiTailwindcss,
  SiGoogleplay,
} from "react-icons/si";
import PropTypes from "prop-types";
const PREFIX = "ProjectCard";

const classes = {
  root: `${PREFIX}-root`,
  githubLinksAnchor: `${PREFIX}-githubLinksAnchor`,
  LanguageChip: `${PREFIX}-LanguageChip`,
};

const StyledCard = styled(Card)(({ theme }) => ({
  [`&.${classes.root}`]: {
    backgroundColor: "#ffffff",
    padding: "1rem",
  },

  [`& .${classes.githubLinksAnchor}`]: {
    "&:visited": {
      color: "#000000",
    },
    "&:link": {
      color: "#000000",
    },
  },

  [`& .${classes.LanguageChip}`]: {
    margin: "0.3rem",
    borderRadius: "5px",
    padding: "0.5rem",
    fontWeight: "bold",
  },
}));

export default function ProjectCard({
  heading,
  description,
  languages,
  linkAriaLabel,
  repoLink,
  siteLink,
  shopLink,
}) {
  //a function that find the correct icon for a given language array
  const getIcon = (language) => {
    switch (language) {
      case "Kotlin":
        return <SiKotlin />;
      case "React Native":
      case "ReactJS":
        return <SiReact />;
      case "Redux toolkit":
        return <SiRedux />;
      case "Tailwind CSS":
        return <SiTailwindcss />;
      case "NextJS":
        return <SiNextdotjs />;
      case "Material UI":
        return <SiMaterialui />;
      case "Styled components":
        return <SiStyledcomponents />;
      case "HTML":
        return <SiHtml5 />;
      case "CSS":
        return <SiCss3 />;
      case "Javascript":
        return <SiJavascript />;
      case "JQuery":
        return <SiJquery />;
      case "Java":
        return <SiJava />;
      case "SQLite":
        return <SiSqlite />;
      default:
        return null;
    }
  };

  return (
    <StyledCard raised className={classes.root}>
      <CardHeader id="heading" title={heading} data-testid="projectHeading" />
      <CardContent>
        <Typography variant="body2">{description}</Typography>
        <Typography variant="overline">
          <br />
          {languages.map((language) => (
            <Chip
              icon={getIcon(language)}
              color="secondary"
              key={language}
              label={language}
              className={classes.LanguageChip}
              data-testid="languageChip"
            />
          ))}
        </Typography>
      </CardContent>
      <CardActions>
        {/* Link to github */}
        <Tooltip title="Github Repo" arrow>
          <IconButton
            data-testid="githublink"
            data-block="projectlink"
            role="button"
            color="secondary"
            aria-label={linkAriaLabel}
            href={repoLink}
            size="large"
          >
            <FiGithub />
          </IconButton>
        </Tooltip>

        {shopLink && (
          <Tooltip title="Play Store Link" arrow>
            <IconButton
              data-testid="shoplink"
              data-block="shoplink"
              role="button"
              color="secondary"
              aria-label={linkAriaLabel}
              href={shopLink}
              size="large"
            >
              <SiGoogleplay />
            </IconButton>
          </Tooltip>
        )}

        {siteLink ? (
          <Tooltip title="Live Site Link" arrow>
            <IconButton
              data-testid="externalSiteLink"
              data-block="projectLinkToSite"
              role="button"
              color="secondary"
              aria-labelledby={heading}
              href={siteLink}
              size="large"
            >
              <FiExternalLink />
            </IconButton>
          </Tooltip>
        ) : null}
      </CardActions>
    </StyledCard>
  );
}

ProjectCard.propTypes = {
  heading: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  languages: PropTypes.arrayOf(PropTypes.string).isRequired,
  linkAriaLabel: PropTypes.string.isRequired,
  repoLink: PropTypes.string.isRequired,
  siteLink: PropTypes.string,
  shopLink: PropTypes.string,
};
