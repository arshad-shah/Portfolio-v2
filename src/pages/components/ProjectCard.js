import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardHeader from "@material-ui/core/CardHeader";
import useStyles from "./ProjectCard.styles";
import { IconButton, Typography } from "@material-ui/core";
import GitHubIcon from "@material-ui/icons/GitHub";
import ShopRoundedIcon from "@material-ui/icons/ShopRounded";

export default function ProjectCard({
  heading,
  description,
  languages,
  linkAriaLabel,
  repoLink,
}) {
  const classes = useStyles();
  return (
    <Card raised className={classes.root}>
      <CardHeader title={heading} />
      <CardContent>
        <Typography variant="body2">{description}</Typography>
        <Typography variant="overline">
          Languages Used: <br />
          {languages}
        </Typography>
      </CardContent>
      <CardActions>
        {/* Link to github */}
        <IconButton role="button" color="default" aria-label={linkAriaLabel}>
          <a className={classes.githubLinksAnchor} href={repoLink}>
            {heading === "Android App Project | Nimaz" ? (
              <ShopRoundedIcon />
            ) : (
              <GitHubIcon />
            )}
          </a>
        </IconButton>
      </CardActions>
    </Card>
  );
}
