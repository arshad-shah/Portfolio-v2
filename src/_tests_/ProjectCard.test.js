import React from 'react';
import { render, screen } from '@testing-library/react';
import ProjectCard from '../components/ProjectCard';

const data = [
	{
		heading: 'Portfolio website version 1.0',
		description:
			'A Portfolio site that contains all the work i have done so far. Includes work from college such as exercises, projects and also from outside college such as the android app.This is the first version written in pure HTML, CSS and javascript with a pinch of jquery.',
		languages: ['HTML', 'CSS', 'Javascript', 'jQuery'],
		linkAriaLabel: 'link to Github for portfolio version 1',
		repoLink: 'https://github.com/arshad-shah/arshad-shah.github.io',
		siteLink: 'https://arshad-shah.github.io/portfoliov1/index.html',
	},
];
describe('ProjectCard of the site:', () => {
	it('should render ProjectCard with all the information in it', () => {
		const { container } = render(
			<ProjectCard
				heading={data[0].heading}
				description={data[0].description}
				languages={data[0].languages}
				linkAriaLabel={data[0].linkAriaLabel}
				repoLink={data[0].repoLink}
				siteLink={data[0].siteLink}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it("should render the heading 'Portfolio website version 1.0'", () => {
		render(
			<ProjectCard
				heading={data[0].heading}
				description={data[0].description}
				languages={data[0].languages}
				linkAriaLabel={data[0].linkAriaLabel}
				repoLink={data[0].repoLink}
				siteLink={data[0].siteLink}
			/>
		);

		const projectHeading = screen.getByText('Portfolio website version 1.0');

		expect(projectHeading).toBeInTheDocument();
	});

	it('should render the description', () => {
		render(
			<ProjectCard
				heading={data[0].heading}
				description={data[0].description}
				languages={data[0].languages}
				linkAriaLabel={data[0].linkAriaLabel}
				repoLink={data[0].repoLink}
				siteLink={data[0].siteLink}
			/>
		);

		const description = screen.getByText(
			'A Portfolio site that contains all the work i have done so far. Includes work from college such as exercises, projects and also from outside college such as the android app.This is the first version written in pure HTML, CSS and javascript with a pinch of jquery.'
		);

		expect(description).toBeInTheDocument();
	});
});
