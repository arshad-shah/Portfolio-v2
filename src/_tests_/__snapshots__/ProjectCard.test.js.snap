// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard of the site: should render ProjectCard with all the information in it 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root ProjectCard-root css-c6x3ll-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      data-testid="projectHeading"
      id="heading"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Portfolio website version 1.0
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        A Portfolio site that contains all the work i have done so far. Includes work from college such as exercises, projects and also from outside college such as the android app.This is the first version written in pure HTML, CSS and javascript with a pinch of jquery.
      </p>
      <span
        class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
      >
        <br />
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary ProjectCard-LanguageChip css-kf42lv-MuiChip-root"
          data-testid="languageChip"
        >
          <svg
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSecondary"
            fill="currentColor"
            height="1em"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M1.5 0h21l-1.91 21.563L11.977 24l-8.564-2.438L1.5 0zm7.031 9.75l-.232-2.718 10.059.003.23-2.622L5.412 4.41l.698 8.01h9.126l-.326 3.426-2.91.804-2.955-.81-.188-2.11H6.248l.33 4.171L12 19.351l5.379-1.443.744-8.157H8.531z"
            />
          </svg>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            HTML
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary ProjectCard-LanguageChip css-kf42lv-MuiChip-root"
          data-testid="languageChip"
        >
          <svg
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSecondary"
            fill="currentColor"
            height="1em"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M1.5 0h21l-1.91 21.563L11.977 24l-8.565-2.438L1.5 0zm17.09 4.413L5.41 4.41l.213 2.622 10.125.002-.255 2.716h-6.64l.24 2.573h6.182l-.366 3.523-2.91.804-2.956-.81-.188-2.11h-2.61l.29 3.855L12 19.288l5.373-1.53L18.59 4.414z"
            />
          </svg>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            CSS
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary ProjectCard-LanguageChip css-kf42lv-MuiChip-root"
          data-testid="languageChip"
        >
          <svg
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSecondary"
            fill="currentColor"
            height="1em"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M0 0h24v24H0V0zm22.034 18.276c-.175-1.095-.888-2.015-3.003-2.873-.736-.345-1.554-.585-1.797-1.14-.091-.33-.105-.51-.046-.705.15-.646.915-.84 1.515-.66.39.12.75.42.976.9 1.034-.676 1.034-.676 1.755-1.125-.27-.42-.404-.601-.586-.78-.63-.705-1.469-1.065-2.834-1.034l-.705.089c-.676.165-1.32.525-1.71 1.005-1.14 1.291-.811 3.541.569 4.471 1.365 1.02 3.361 1.244 3.616 2.205.24 1.17-.87 1.545-1.966 1.41-.811-.18-1.26-.586-1.755-1.336l-1.83 1.051c.21.48.45.689.81 1.109 1.74 1.756 6.09 1.666 6.871-1.004.029-.09.24-.705.074-1.65l.046.067zm-8.983-7.245h-2.248c0 1.938-.009 3.864-.009 5.805 0 1.232.063 2.363-.138 2.711-.33.689-1.18.601-1.566.48-.396-.196-.597-.466-.83-.855-.063-.105-.11-.196-.127-.196l-1.825 1.125c.305.63.75 1.172 1.324 1.517.855.51 2.004.675 3.207.405.783-.226 1.458-.691 1.811-1.411.51-.93.402-2.07.397-3.346.012-2.054 0-4.109 0-6.179l.004-.056z"
            />
          </svg>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Javascript
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary ProjectCard-LanguageChip css-kf42lv-MuiChip-root"
          data-testid="languageChip"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            jQuery
          </span>
        </div>
      </span>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-khne5s-MuiCardActions-root"
    >
      <a
        aria-label="link to Github for portfolio version 1"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeLarge css-wrxors-MuiButtonBase-root-MuiIconButton-root"
        data-block="projectlink"
        data-mui-internal-clone-element="true"
        data-testid="githublink"
        href="https://github.com/arshad-shah/arshad-shah.github.io"
        role="button"
        tabindex="0"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        aria-label="Live Site Link"
        aria-labelledby="Portfolio website version 1.0"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeLarge css-wrxors-MuiButtonBase-root-MuiIconButton-root"
        data-block="projectLinkToSite"
        data-mui-internal-clone-element="true"
        data-testid="externalSiteLink"
        href="https://arshad-shah.github.io/portfoliov1/index.html"
        role="button"
        tabindex="0"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
